bin/autodetect/Main.cc

```diff
@@ -152,6 +151,12 @@ int main(int argc, char** argv) {
     // statically links its own version library.
     LOG_DEBUG(<< ml::ver::CBuildInfo::fullInfo());
 
+    if (unknownTokens.size()) {
+        LOG_ERROR(<< "Unkown command line options: "
+                  << ml::core::CContainerPrinter::print(unknownTokens));
+        return EXIT_FAILURE;
+    }
```

Comment:
You won't need this check if you configure Boost Program Options to ban unknown arguments.